task-impls/src/network.rs

```diff
@@ -77,57 +84,102 @@ impl<
+                            return;
+                        }
+                        _ => {
+                            error!("Got unexpected message type in network task!");
+                            return;
+                        }
+                    },
                     _ => {
-                        error!("Got unexpected message type in network task!");
                         return;
                     }
                 },
-                Either::Right(committee_message) => match committee_message {
-                    CommitteeConsensusMessage::DAProposal(proposal) => {
-                        SequencingHotShotEvent::DAProposalRecv(proposal.clone(), sender)
-                    }
-                    CommitteeConsensusMessage::DAVote(vote) => {
-                        SequencingHotShotEvent::DAVoteRecv(vote.clone())
-                    }
-                    CommitteeConsensusMessage::DACertificate(cert) => {
-                        SequencingHotShotEvent::DACRecv(cert)
-                    }
-                },
+                MessageKind::Data(message) => {
+                    return;
+                }
+                MessageKind::_Unreachable(_) => unimplemented!(),
             },
-            MessageKind::Data(message) => {
-                match message {
-                    // ED Why do we need the view number in the transaction?
-                    hotshot_types::message::DataMessage::SubmitTransaction(
-                        transaction,
-                        view_number,
-                    ) => {
-                        SequencingHotShotEvent::TransactionRecv(transaction)
+            NetworkTaskKind::Committee => {
+                match message.kind {
+                    MessageKind::Consensus(consensus_message) => match consensus_message.0 {
+                        Either::Left(general_message) => match general_message {
+                            GeneralConsensusMessage::Proposal(_)
+                            | GeneralConsensusMessage::Vote(_)
+                            | GeneralConsensusMessage::ViewSyncVote(_)
+                            | GeneralConsensusMessage::ViewSyncCertificate(_) => {
+                                return;
+                            }
+                            _ => {
+                                error!("Got unexpected message type in network task!");
+                                return;
+                            }
+                        },
+                        Either::Right(committee_message) => match committee_message {
+                            CommitteeConsensusMessage::DAProposal(proposal) => {
+                                SequencingHotShotEvent::DAProposalRecv(proposal.clone(), sender)
+                            }
+                            CommitteeConsensusMessage::DAVote(vote) => {
+                                SequencingHotShotEvent::DAVoteRecv(vote.clone())
```

Comment:
does this need to be clone?

Reply:
No it doesn't! I've also removed another `clone` above it.